src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseTaskService.java

```diff
@@ -121,13 +121,16 @@ public List<ProgrammingExerciseTask> getSortedTasks(ProgrammingExercise exercise
     }
 
     /**
-     * Gets all tasks of an exercise
+     * Gets all tasks of an exercise excluding inactive test cases
      *
      * @param exerciseId of the programming exercise
      * @return Set of all tasks and its test cases
      */
-    public Set<ProgrammingExerciseTask> getTasks(long exerciseId) {
-        return programmingExerciseTaskRepository.findByExerciseIdWithTestCaseAndSolutionEntriesElseThrow(exerciseId);
+    public Set<ProgrammingExerciseTask> getTasksWithoutInactive(long exerciseId) {
```

Comment:
```suggestion
    public Set<ProgrammingExerciseTask> getTasksWithoutInactiveTestCases(long exerciseId) {
```
(nitpick) It’s not the tasks themself that are inactive, the inactive test cases associated with the task are filtered.